### DISCUSSION TITLE
Driver not working for me - NEED HELP

### DISCUSSION THREAD

**Discussion by Zeus5753** [Louisvdw/dbus-serialbattery#358] [Q&A] [answered]
I followed the installation procedures and this is the error I ge
![image](https://user-images.githubusercontent.com/101138658/208193556-b7d295c7-bb0a-4e71-b089-f4da8d175fa9.png)

Please help

**Comment by cm2962** [comment]
I have the same issue since I was trying to update to 0.142

**Reply by Zeus5753** [reply]
Do older versions work?

**Reply by cm2962** [reply]
Yes. I downgraded to 0.14.1 and everything is good again

**Reply by Zeus5753** [reply]
Great to know! Sadly it was on a client's project that's over an hour drive from me so I guess i'll fix it next time I'm there. I tried to setup remote access but forgot to setup port forwarding on his router. Was pressed for time...

**Comment by Solarbastler** [comment]
Hello, what should the command be to install an older version via SSH on Raspberry.
wget https://github.com/Louisvdw/dbus-serialbattery/releases/tag/v0.14   ...?

I don't know enough about it :)

**Reply by Louisvdw** [reply]
Go to the older release in github and right click on the venus-data.tar.gz, select Copy link address and paste that after wget 
So
> wget https://github.com/Louisvdw/dbus-serialbattery/releases/download/v0.14.1/venus-data.tar.gz

**Reply by Solarbastler** [reply]
Thx , it works 👍

**Comment by aleksbyhun** [comment]
I figured out how to fix this. Comment the row # 27 "from revov import Revov" in file /data/etc/dbus-serialbattery/dbus-serialbattery.py
Remove the USB stick if you installed using this option. Reboot the GX device. Done.

**Comment by JimS-JJ** [comment]
Hi LouisVDW, Great work you are doing for this community, Thanks.  I have a similar issue, using a Cerbo GX V2.92, Driver v0.14.3, removed the # for Sinowealth, rebooted and still not able to connect. I did notice when I plugged a USB tester into the Cerbo it showed 5 volts, and I have tested the Daly UART voltage and it is working with 3.3 volts, although it is NOT connected to the UART.  Could this be causing the problem?  Thanks again.

![Screenshot 2023-01-03 151423](https://user-images.githubusercontent.com/121907716/210457876-ed43cb99-b0e1-44c3-8643-6893c4fd606b.png)

**Reply by Louisvdw** [reply]
@JimS-JJ if the BMS is not picked up it is most often an issue with the connection or wiring. Make sure that you have the correct adapter for the port you are using. 
The Sinowealth chip is mostly used for the smaller BMS models. If you have one of them it is also possible that your chip model is not suported in the driver as they do have many different chips they use (but most are supported)

**Comment by JimS-JJ** [comment, marked as answer]
My Friend, I just found the answer this morning.  On the Daly BMS 250 amp, mine has a monitor output, a 3 pin connector that Daly defines that as being a Sinowealth chipset.  I have used that before to connect to my computer running the Sinowealth software.  Daly's cable that is a 3 pin to USB says "monitor/USB".  That cable was plugged into an isolator and into the CerboGX this morning and I started getting data.  For those that are trying the UART connector, 6 pin on the opposite side need to use this monitor/USB 3 pin connector and it will work.  Please pass on this information.  Thanks.  Very relieved, Jim  1/4/22
